src/server/services/varen/varen-notifications.ts

```diff
@@ -0,0 +1,122 @@
+    link: {
+      to: AppRoutes.VAREN,
+      title: 'Bekijk details',
+    },
+  };
+}
+
+function createVarenNotifications(
+  zaak: VarenVergunningFrontend
+): MyNotification[] {
+  // We do not link to or show processed aanvragen, only vergunningen
+  const ctaLinkToThemaOrDetail =
+    !zaak.processed || isVergunning(zaak)
+      ? generatePath(AppRoutes['VAREN/DETAIL'], {
+          id: zaak.id,
+          caseType: zaak.caseType,
+        })
+      : AppRoutes.VAREN;
+
+  const notifications = zaak.steps
+    .filter((step) => step.isChecked)
+    .map((step) => {
+      const baseNotification = {
+        datePublished: step.datePublished,
+        thema: Themas.VAREN,
+        link: {
+          to: ctaLinkToThemaOrDetail,
+          title: 'Bekijk details',
+        },
+      };
+      switch (step.status) {
+        case 'Ontvangen':
+          return {
+            ...baseNotification,
+            id: `varen-${zaak.id}-ontvangen-notification`,
+            title: `Aanvraag ${zaak.caseType} ontvangen`,
+            description: `Wij hebben uw aanvraag ontvangen.`,
+          };
+        case 'In behandeling':
+          return {
+            ...baseNotification,
+            id: `varen-${zaak.id}-inbehandeling-notification`,
+            title: `Aanvraag ${zaak.caseType} in behandeling`,
+            description: `Wij hebben uw aanvraag in behandeling genomen.`,
+          };
+        case 'Meer informatie nodig':
+          return {
+            ...baseNotification,
+            id: `varen-${zaak.id}-meerinformatienodig-notification`,
+            title: `Meer informatie nodig omtrent uw ${zaak.caseType} aanvraag`,
+            description: `Er is meer informatie nodig om de aanvraag verder te kunnen verwerken.`,
+          };
+        case 'Besluit':
+          return {
+            ...baseNotification,
+            id: `varen-${zaak.id}-afgehandeld-notification`,
+            title: `Aanvraag ${zaak.caseType} afgehandeld`,
+            description: `Wij hebben uw aanvraag afgehandeld.`,
+          };
+      }
+    });
+
+  // If datePublished of notifications are equal, the last notification is shown first
+  return notifications.reverse();
+}
+
+export async function fetchVarenNotifications(
+  requestID: RequestID,
+  authProfileAndToken: AuthProfileAndToken
+) {
+  const varenResponse = await fetchVaren(requestID, authProfileAndToken);
+
+  if (varenResponse.status === 'ERROR') {
+    return apiDependencyError({ varen: varenResponse });
+  }
+  const notifications = [];
+
+  const rederRegistration = varenResponse.content.find(
+    (zaak) => zaak.caseType === 'Varen registratie reder'
+  );
+  if (rederRegistration) {
+    notifications.push(
+      createVarenRederRegisteredNotification(rederRegistration)
+    );
+  }
+
+  const zaken = varenResponse.content.filter(
+    (zaak) => zaak.caseType !== 'Varen registratie reder'
+  );
+  notifications.push(...zaken.map(createVarenNotifications).flat());
```

Comment:
Hier pak je alleen het laatste element toch? Is iets als het volgende dan niet leesbaarder?
```suggestion
const notifications = zaken.map(createVarenNotifications).flat();
notifications.push(notifications.at(-1));
```

Reply:
Willen we niet alle historische meldingen ook tonen of tonen we alleen meldingen die op dit moment relevant zijn voor de gebruiker?

Bij varen hebben we bijvoorbeeld dat een goedgekeurde aanvraag verdwijnt (wijzigingen aan de hand van de goedgekeurde aanvraag komen terecht in de vergunning, wat een ander object/pagina is). Dan lijkt het mij wel fijn om hiervan een melding te tonen van elke stap die deze aanvraag is doorlopen.

Reply:
Sorry mijn fout Terry. Ik was nieuwsgierig wat het deed en keek even in de interpreter:
![image](https://github.com/user-attachments/assets/fcdde6e5-4c8f-449c-9e00-18dc58a3e1a6)
.... maar push returned de length natuurlijk en ik dacht oh drie dat is het laatste element...